### PR TITLE
Update DG for implementation of EditCommand

### PR DESCRIPTION
**Pull request by tohyuting**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -217,7 +217,17 @@ _{more aspects and alternatives to be added}_
...
...f two types of `ct/COMMAND_TYPE` is provided, the last type specified will use processed as the type selected by user. This applies for other prefixes used as inputs as well. The code will throw a **ParseException** if no compulsory prefixes or only one of the compulsory prefixes are given. Afterwards, **EditCommandParser** will check to see if at least one field is provided for editing. The code will throw a **ParseException** if no field for editing of suppliers or warehouses is provided. Parsing of user input then begins. If there are any inappropriate fields supplied (e.g. input a string for index or phone prefix), **ParseException** will be thrown. Furthermore, **EditCommandParser** will also throw a **ParseException** if the fields input result in no changes to the existing supplier or warehouse entry. Finally, **EditCommandParser** will create a new **EditCommand** to be executed. **EditCommand** takes in an index for supplier or warehouse list and an **EditDescriptor** containing the fields to be edited when the command is executed. The edited supplier or warehouse will be updated in the model, allowing users to see the changes done for the respective supplier or warehouse.
```
Comment (jeffreytjs) [inline docs/DeveloperGuide.md]: In the 3rd sentence:
- Should add a space between "The" and `edit`

In the 4th sentence:
- Could be two different way of expression in "the last type specified will use processed"

The rest are well described.

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -217,7 +217,17 @@ _{more aspects and alternatives to be added}_
...
... remind users in case they made a minor mistake, resulting in a supplier or warehouse to not update the way they intended for it to. Lastly, a command type prefix, `ct/COMMAND_TYPE` is required in the implementation of `edit` command to indicate whether user wishes to edit a warehouse or supplier entry. Without a command type prefix, an alternative would be for a `TYPE` parameter, where user have to indicate `supplier` or `warehouse`. However, this may not be suitable for our target user, who wishes to update stocks quickly. Hence, our team decided to use a command type prefix in place of a parameter. Furthermore, another alternative considered would be to create separate commands for warehouse and suppliers respectively. For example, `editw` and `edits` to represent edit warehouse and edit supplier. However, this might increase duplicated codes, since minimal changes to the code would be found for each class of command. By using a prefix, this helps us to reduce any potential code duplication. Therefore, our team decided to implement edit command by taking in prefixes and throwing our relevant exceptions at appropriate points after considering code quality and end user experience.
```
Comment (jeffreytjs) [inline docs/DeveloperGuide.md]: I cant figure out why this paragraph is bold in the file. (Can click the three dots at the top right then click `view file`)

Minor issue:
Can make warehouse plural as well in this sentence - "Furthermore, another alternative considered would be to create separate commands for warehouse and suppliers respectively."

**Review by jeffreytjs — COMMENTED** [review]
Well described with minor proposed edits

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (tohyuting) [reply docs/DeveloperGuide.md]: Thanks for pointing them out! And yes, I think I made a typo on the 4th sentence, will change it to "will be used to process user input"

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (tohyuting) [reply docs/DeveloperGuide.md]: I think I figured out why, there was not space between the last line of the paragraph and the line right after it (with -), will fix this in my next commit!

Will also fix the grammar as well, thanks for pointing that out.

**Review by jeffreytjs — APPROVED** [review]
LGTM

**Pull request merged**